### PR TITLE
Add Lighthouse charts

### PR DESCRIPTION
**Pull request by vthapar**
This adds the deployment for lighthouse-agent, service account and setting up required RBAC.

### CODE REVIEW

File: submariner/templates/lighthouse.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        - command:
+            - lighthouse-agent.sh
+          env:
+            - name: SUBMARINER_NAMESPACE
+              value: "{{ .Release.Namespace }}"
+            - name: SUBMARINER_CLUSTERID
+              value: "{{ .Values.submariner.clusterId }}"
+            - name: SUBMARINER_DEBUG
+              value: "{{ .Values.submariner.debug }}"
+            - name: SUBMARINER_BROKER
```
Comment (tpantelis) [inline submariner/templates/lighthouse.yaml:37]: SUBMARINER_BROKER isn't used/needed.

File: submariner/templates/crd.yaml [hunk 2]
```diff
@@ -40,4 +40,34 @@ spec:
     kind: Gateway
     plural: gateways
   scope: Namespaced
+---
```
Comment (mkolesnik) [inline submariner/templates/crd.yaml:43]: I guess you could have this all optional with `{{- if .Values.submariner.serviceDiscovery }}`

File: submariner-k8s-broker/templates/crd.yaml [hunk 3]
```diff
@@ -26,4 +26,34 @@ spec:
     kind: Endpoint
     plural: endpoints
   scope: Namespaced
+---
```
Comment (mkolesnik) [inline submariner-k8s-broker/templates/crd.yaml:29]: I guess you could have this all optional with `{{- if .Values.submariner.serviceDiscovery }}`

File: submariner/templates/lighthouse.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      - command:
+        - lighthouse-agent.sh
+        env:
+        - name: SUBMARINER_NAMESPACE
+          value: "{{ .Release.Namespace }}"
+        - name: SUBMARINER_CLUSTERID
+          value: "{{ .Values.submariner.clusterId }}"
+        - name: SUBMARINER_DEBUG
+          value: "{{ .Values.submariner.debug }}"
+{{- if eq .Values.broker.type "k8s" }}
```
Comment (mkolesnik) [inline submariner/templates/lighthouse.yaml:34]: I don't think this is necessary. If someone wants to add support of lighthouse via a different broker then he can add this (and all other necessary changes) later

File: submariner/templates/svc-acct.yaml [hunk 5]
```diff
@@ -33,3 +33,15 @@ metadata:
     chart: {{ template "submariner.chart" . }}
     app: {{ template "submariner.name" . }}
 {{- end }}
+---
+  {{- if .Values.serviceAccounts.lighthouse.create }}
```
Comment (mkolesnik) [inline submariner/templates/svc-acct.yaml:37]:
```suggestion
{{- if .Values.serviceAccounts.lighthouse.create }}
```

File: submariner/templates/svc-acct.yaml [hunk 6]
```diff
@@ -33,3 +33,15 @@ metadata:
...
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "submariner.lighthouseServiceAccountName" . }}
+  labels:
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+    chart: {{ template "submariner.chart" . }}
+    app: {{ template "submariner.name" . }}
+  {{- end }}
```
Comment (mkolesnik) [inline submariner/templates/svc-acct.yaml:47]:
```suggestion
{{- end }}
```

File: submariner/values.yaml [hunk 7]
```diff
@@ -58,6 +58,11 @@ globalnet:
     repository: submariner-globalnet
     tag: local
     pullPolicy: IfNotPresent
+lighthouseAgent:
```
Comment (mkolesnik) [inline submariner/values.yaml:61]:
```suggestion
lighthouse:
```

File: submariner/templates/lighthouse.yaml [hunk 8]
```diff
@@ -0,0 +1,54 @@
+{{- if .Values.submariner.serviceDiscovery }}
+apiVersion: apps/v1
+kind: Deployment
```
Comment (mkolesnik) [inline submariner/templates/lighthouse.yaml:3]: Shouldn't this be a DaemonSet like the other subm resources?

File: submariner/templates/crd.yaml (hunk 2 above)
Comment (vthapar) [reply submariner/templates/crd.yaml:43]: Yes. Thoghth so but went with this coz this is what we last discussed. Makes sense.

File: submariner/templates/lighthouse.yaml (hunk 4 above)
Comment (vthapar) [reply submariner/templates/lighthouse.yaml:34]: Cool. makes things easier and more similar to way we have in operator.

File: submariner/templates/lighthouse.yaml (hunk 8 above)
Comment (vthapar) [reply submariner/templates/lighthouse.yaml:3]: No, we need only one pod running agent on any node. It doesn't have any networking pieces.

File: submariner/templates/lighthouse.yaml [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  selector:
+    matchLabels:
+      app: {{ template "submariner.fullname" . }}-lighthouse-agent
+  template:
+    metadata:
+      labels:
+        app: {{ template "submariner.fullname" . }}-lighthouse-agent
+    spec:
+      serviceAccount: submariner:lighthouse
+      serviceAccountName: submariner-lighthouse
```
Comment (mkolesnik) [inline submariner/templates/lighthouse.yaml:23]: Should this use `{{ template "submariner.lighthouseServiceAccountName" . }}`?

File: submariner/templates/lighthouse.yaml [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ template "submariner.fullname" . }}-lighthouse-agent
+  template:
+    metadata:
+      labels:
+        app: {{ template "submariner.fullname" . }}-lighthouse-agent
+    spec:
+      serviceAccount: submariner:lighthouse
```
Comment (mkolesnik) [inline submariner/templates/lighthouse.yaml:22]: Is this really necessary given that you have `serviceAccountName`?

File: submariner/templates/_helpers.tpl [hunk 11]
```diff
@@ -63,3 +63,14 @@ Create the name of the submariner-globalnet service account to use
     {{ default "default" .Values.serviceAccounts.globalnet.name }}
 {{- end -}}
 {{- end -}}
+
+{{/*
+Create the name of the submariner-lighthouse service account to use
+*/}}
+{{- define "submariner.lighthouseServiceAccountName" -}}
+{{- if .Values.serviceAccounts.lighthouse.create -}}
```
Comment (mkolesnik) [inline submariner/templates/_helpers.tpl:71]: I think you can safely use `submariner.serviceDiscovery` here for consistency

File: submariner/templates/lighthouse.yaml (hunk 10 above)
Comment (vthapar) [reply submariner/templates/lighthouse.yaml:22]: Not needed. Carry over from old yamls where we were still using serviceAccount.

File: submariner/templates/lighthouse.yaml (hunk 9 above)
Comment (vthapar) [reply submariner/templates/lighthouse.yaml:23]: Yes. it should.

File: submariner/templates/_helpers.tpl (hunk 11 above)
Comment (vthapar) [reply submariner/templates/_helpers.tpl]: done

File: submariner/templates/lighthouse.yaml (hunk 10 above)
Comment (vthapar) [reply submariner/templates/lighthouse.yaml]: done

**Pull request merged**